L.Control.Geonames.js

```diff
@@ -91,6 +100,50 @@ L.Control.Geonames = L.Control.extend({
             callbackName
         );
     },
+    _getSearchParams: function() {
+        var params = '';
+
+        for (var paramName in this.options.adminCodes) {
+            if (ADMIN_CODES.indexOf(paramName) == -1) {
+                continue;
+            }
+
+            var paramValue = this.options.adminCodes[paramName];
+
+            if (typeof paramValue == 'function') {
+                paramValue = paramValue();
+                if (paramValue) {
+                    params += '&' + paramName + '=' + paramValue;
+                }
+            } else if (typeof paramValue == 'string' && paramValue.length > 0) {
+                params += '&' + paramName + '=' + paramValue;
+            }
+        }
+
+        var bbox = this._getBbox();
+        if (bbox) {
+            params += bbox;
+        }
+
+        return params;
+    },
+    _getBbox: function() {
+        if (this._bbox) {
```

Comment:
@ka7eh I don't think you need to cache this result, and it might get in the way if `this.options.bbox` is getting dynamically updated.